src/Testcontainers.Typesense/TypesenseConfiguration.cs

```diff
@@ -0,0 +1,69 @@
+namespace Testcontainers.Typesense;
+
+public class TypesenseConfiguration : ContainerConfiguration
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseConfiguration" /> class.
+    /// </summary>
+    public TypesenseConfiguration(
+      int port = TypesenseBuilder.DefaultPort,
+      string apiKey = TypesenseBuilder.DefaultApiKey,
+      bool enableCors = TypesenseBuilder.DefaultCors,
+      string volume = TypesenseBuilder.DefaultVolume)
+    {
+        Port = port;
+        ApiKey = apiKey;
+        EnableCors = enableCors;
+        Volume = volume;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseConfiguration" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    public TypesenseConfiguration(IResourceConfiguration<CreateContainerParameters> resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        // Passes the configuration upwards to the base implementations to create an updated immutable copy.
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseConfiguration" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    public TypesenseConfiguration(IContainerConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        // Passes the configuration upwards to the base implementations to create an updated immutable copy.
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseConfiguration" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    public TypesenseConfiguration(TypesenseConfiguration resourceConfiguration)
+        : this(new TypesenseConfiguration(), resourceConfiguration)
+    {
+        // Passes the configuration upwards to the base implementations to create an updated immutable copy.
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseConfiguration" /> class.
+    /// </summary>
+    /// <param name="oldValue">The old Docker resource configuration.</param>
+    /// <param name="newValue">The new Docker resource configuration.</param>
+    public TypesenseConfiguration(TypesenseConfiguration oldValue, TypesenseConfiguration newValue)
```

Comment:
Doesn't this constructor need to take care of `EnableCors` and `Volume` as well?